File: Carrito-de-Compras/02-Fase2-POO-BlueJ/Proyecto-BlueJ/Producto.java
public class Producto
{
    private int idProducto;
    private String nombre;
    private String descripcion;
    private double precio;

    public Producto(int idProducto, String nombre, String descripcion, double precio)
    {
        this.idProducto = idProducto;
        this.nombre = nombre;
        this.descripcion = descripcion;
        this.precio = precio;
    }

    public int getIdProducto()
    {
        return idProducto;
    }

    public String getNombre()
    {
        return nombre;
    }

    public String getDescripcion()
    {
        return descripcion;
    }

    public double obtenerPrecio()
    {
        return precio;
    }

    public void mostrarInformacion()
    {
        System.out.println("ID: " + idProducto);
        System.out.println("Producto: " + nombre);
        System.out.println("Descripcion: " + descripcion);
        System.out.println("Precio: Q" + precio);
    }
}

File: Carrito-de-Compras/03-Fase3-CSharp/Models/Producto.cs
namespace CarritoCompras.Models;

public class Producto
{
    public int IdProducto { get; private set; }
    public string Nombre { get; private set; }
    public string Descripcion { get; private set; }
    public decimal Precio { get; private set; }
    public int Stock { get; private set; }

    public Producto(int idProducto, string nombre, string descripcion, decimal precio, int stock)
    {
        IdProducto = idProducto;
        Nombre = nombre;
        Descripcion = descripcion;
        Precio = precio;
        Stock = stock;
    }

    public bool HayStock(int cantidad)
    {
        return cantidad > 0 && Stock >= cantidad;
    }

    public bool DescontarStock(int cantidad)
    {
        if (!HayStock(cantidad))
            return false;

        Stock -= cantidad;
        return true;
    }

    public void AumentarStock(int cantidad)
    {
        if (cantidad > 0)
            Stock += cantidad;
    }

    public void MostrarInformacion()
    {
        Console.WriteLine($"{IdProducto}. {Nombre} | Q{Precio:0.00} | Stock: {Stock}");
        Console.WriteLine($"   {Descripcion}");
    }
}
